public/assets/scripts/blocklyflasher.js

```diff
@@ -0,0 +1,1790 @@
+        this.eventManager.addListener(type, listener);
+    };
+
+
+    this.pluginHandler = new function(){
+
+        this.owner;
+
+        this.serialMonitorPort = null;
+
+        this.max_monitor_length = 5000;
+
+        this.uuid4 = function () {
+            var d = new Date().getTime();
+            var uuid = 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, function(c) {
+                var r = (d + Math.random()*16)%16 | 0;
+                d = Math.floor(d/16);
+                return (c=='x' ? r : (r&0x7|0x8)).toString(16);
+            });
+            return uuid;
+        };
+
+        this.tabID = this.uuid4();
+
+        this.currentPorts = [];
+
+        this.doflashBootloader = function(programmer, board)
+        {
+            document.getElementById('plugin0').flashBootloader(
+                (this.portslist.selectedIndex == -1 || programmer['communication'] != 'serial')?'':this.portslist.options[this.portslist.selectedIndex].text,
+                programmer['protocol'],
+                programmer['communication'],
+                programmer['speed'],
+                programmer['force'],
+                programmer['delay'],
+                board['bootloader']['high_fuses'],
+                board['bootloader']['low_fuses'],
+                (typeof board['bootloader']['extended_fuses'] === "undefined")?'':board['bootloader']['extended_fuses'],
+                (typeof board['bootloader']['unlock_bits'] === "undefined")?'':board['bootloader']['unlock_bits'],
+                (typeof board['bootloader']['lock_bits'] === "undefined")?'':board['bootloader']['lock_bits'],
+                board['build']['mcu'],
+                bootloader_callback
+            );
+        };
+
+        this.clickedPort = function()
+        {
+            var port = $("#cb_cf_ports").val();
+            var actionId = 43;
+            var metaData = {
+                "selectedPort": port,
+                "tabID": this.tabID
+            };
+            createLogFlasher(actionId, metaData);
+        };
+
+        this.oldPort = null;
+
+        this.logPorts = function () {
+            var oldPort;
+            if (this.oldPort == null)
+                oldPort = '';
+            else
+                oldPort = this.oldPort;
+
+            var newPort;
+            if (this.currentPorts.length > 0)
+                newPort = $("#cb_cf_ports option:selected").text();
+            else
+                newPort = '';
+
+            this.oldPort = newPort;
+
+            if (oldPort != newPort) {
+                var actionId = 38;
+                var metaData = {
+                    "oldPort": oldPort,
+                    "newPort": newPort,
+                    "tabID": this.tabID
+                };
+                createLogFlasher(actionId, metaData);
+            }
+        };
+
+        this.savePort = function() {
+            var cb = this;
+            if(typeof Lawnchair !== "undefined")
+            {
```

Comment:
It's always best to be consistent with your coding style. In this document you have `if`s defined as:
```js
if (something) {
    // do something
}```
,
```js
if (somethingElse){ //note no space after )
//or this
if(somethingElse) { //note no space between if and (
```
and 
```js
if (another)
{
    //something
}
```

Of all those I suggest sticking with the first, it is visually consistent with PHP's PSR-2 coding style which means if you switch between the two it won't be mentally confusing at all.